tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py

```diff
@@ -0,0 +1,919 @@
+
+    # Create hosttemplate and host
+    ht_create_args = {
+        "name": random_str(),
+        "driver": "digitalocean",
+        "secretValues": {
+            "digitaloceanConfig": {
+                "accessToken": do_access_key
+            }
+        }
+    }
+
+    ht0 = client.wait_success(client.create_hostTemplate(**ht_create_args))
+    base_host= create_host(ht0.id, client)
+    base_host_name = base_host["hostname"]
+
+    # (1-1): Create webhook for scale up hosts
+    data = {
+        "name": random_str(),
+        "driver": "scaleHost",
+        "scaleHostConfig": {
+            "action": "up",
+            "amount": 2,
+            "hostTemplateId": base_host.hostTemplateId,
+            "min": 1,
+            "max": 4
+        }
+    }
+
+    resp = create_webhook(env.id, data)
+    assert resp.status_code == 200
+    assert resp.url is not None
+    json_resp = json.loads(resp.content)
+    webhook_up_url = json_resp["url"]
+    webhook_up_id = json_resp["id"]
+
+    print "Webhook is " + repr(webhook_up_url)
+    print "Id is " + repr(webhook_up_id)
+
+    # Execute Webhook and verify that the scale is incremented by
+    # the amount specified
+    wh_resp = requests.post(webhook_up_url)
+    assert wh_resp.status_code == 200
+
+    hosts = client.list_host()
+    host_scale_group = []
+    for host in hosts:
+        if host["hostTemplateId"] == base_host["hostTemplateId"]:
+            host = client.wait_success(host, timeout=DEFAULT_MACHINE_TIMEOUT)
+            host_scale_group.append(host)
+    assert len(host_scale_group) == 3
+
+    host_scale_group = sorted(host_scale_group, key=lambda x: x["created"])
+
+    middle_host = host_scale_group[1]
+    last_host = host_scale_group[2]
+
+    # Checking the name of the hosts
+    base_host_name = base_host_name.split('.')[0]
+    base_host_prefix = base_host_name.rstrip('0123456789')
+    base_host_suffix = base_host_name.split(base_host_prefix)[1]
+
+    for host in host_scale_group:
+        if host["id"] == base_host["id"]:
+            continue
+        current_host_name = host["hostname"]
+        current_host_name = current_host_name.split('.')[0]
+        current_host_prefix = current_host_name.rstrip('0123456789')
+        current_host_suffix = current_host_name.split(current_host_prefix)[1]
+
+        assert current_host_prefix == base_host_prefix
+
+        if base_host_suffix != "":
+            assert string.atoi(current_host_suffix) == string.atoi(base_host_suffix) + 1
+        else:
+            assert current_host_suffix == "2"
+
+        base_host_suffix = current_host_suffix
+
+    # (1-2): Execute Webhook again and ensure the scale
+    # cannot be incremented beyond max
+    wh_resp = requests.post(webhook_up_url)
+    assert wh_resp.status_code == 400
+    json_resp = json.loads(wh_resp.content)
+    print "JSON response is:"
+    print json_resp
+
+    expected_response = "Error Cannot scale above provided max " \
```

Comment:
asserting on exact error messages is brittle and error prone. It is sufficient to assert on just the status_code